From 11287e071393500812ef2f9a3c0729605d96a967 Mon Sep 17 00:00:00 2001
From: Ralf Gommers <ralf.gommers@gmail.com>
Date: Mon, 8 May 2023 17:13:58 +0200
Subject: [PATCH 8/9] BLD: add numpy and python include dirs to propack static
 library

This is needed when building with the g77 ABI wrappers, because
`npy_common.h` is used within them (for the `npy_int64` symbol).
It looks like removing that symbol and replacing it with `int64_t`
isn't completely straightforward; those two symbols are not
compatible.

[skip circle] [skip cirrus]
---
 scipy/sparse/linalg/_propack/setup.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/scipy/sparse/linalg/_propack/setup.py b/scipy/sparse/linalg/_propack/setup.py
index 94a4f055b..fdc67fd34 100644
--- a/scipy/sparse/linalg/_propack/setup.py
+++ b/scipy/sparse/linalg/_propack/setup.py
@@ -1,4 +1,5 @@
 import pathlib
+from distutils.sysconfig import get_python_inc
 
 import numpy as np
 
@@ -65,7 +66,12 @@ def configuration(parent_package='', top_path=None):
         config.add_library(propack_lib,
                            sources=src,
                            macros=cmacros,
-                           include_dirs=src_dir,
+                           include_dirs=[
+                               src_dir,
+                               # because npy_common.h is used in g77 abi wrappers
+                               get_python_inc(),
+                               np.get_include(),
+                           ],
                            depends=['setup.py'])
         ext = config.add_extension(f'_{prefix}propack',
                                    sources=f'{prefix}propack.pyf',
